src/Controller/TwigAwareController.php

```diff
@@ -65,14 +75,33 @@ protected function renderTemplate($template, array $parameters = [], ?Response $
             $template = $this->twig->resolveTemplate($templates);
         }
 
+        // Render the template
         $content = $this->twig->render($template, $parameters);
 
+        // Make sure we have a Response
         if ($response === null) {
             $response = new Response();
         }
-
         $response->setContent($content);
 
-        return $response;
+        // Process the snippet Queue on the Response
+        return $this->widgets->processQueue($response);
+    }
+
+    public function registerBoltWidgets(): void
```

Comment:
this and processQueue should not be parts of TwigAwareController (extending this class is highly optional!) move it to subscriber

Reply:
I've added a `WidgetsSubscriber`, and removed them from `TwigAwareController`. Much better!